fahrschein/src/main/java/org/zalando/fahrschein/ProblemHandlingRequest.java

```diff
@@ -113,15 +113,15 @@ private void handleAuthError(final JsonNode rootNode, final int statusCode) thro
 
     private void handleBatchItemResponse(JsonNode rootNode) throws IOException {
```

Comment:
is it called only when 207 happens or 422 as well ?

Reply:
Both. ProblemHandlingRequest line 38: `if (statusCode == 207 || statusCode >= 400) {`

Reply:
ideally it would be different type of exception, but it creates a breaking change in the api as I understand. I would say just keep it like this for now (do not need to extend api with additional flag if users will start to use it will be more work to get rid of it) because it keeps the behaviour the same in this part. it is probably good idea to change this in the future.

Reply:
we're in 2.0.0-RC phase, it's ok to introduce breaking changes now.

Reply:
in that case I would go with different exceptions e.g. `EventPersistenceException` and `EventValidationException` because this is the way how it is done at the moment in the library (the way of exceptions). It can be easily added to the documentation of the method.